### PR TITLE
feat(notes): add listening session rollover telemetry

### LINKED ISSUE #105: [Feature] Listening Sessions: Voice capture with multi-stage AI synthesis
## Epic
Top-priority initiative to make voice-first note capture + synthesis a core product loop.

## Why
Typing notes during reading breaks flow. Readers need to speak freely (including rambling thoughts), keep reading, and still get useful outputs tied to the current book.

## Product Requirements (MVP)
- Start recording from book detail page with a prominent CTA.
- Active recording state is visually obvious.
- Real-time STT during capture.
- Support long sessions (>=30 minutes).
- If cap is enforced, provide:
  - audible warning
  - prominent visual warning
  - auto-process current session at cutoff
  - forced start of a new session for continuation
- Always save raw transcript as a note.
- Generate synthesized artifacts in book context:
  - insights
  - open questions
  - quotes
  - follow-up questions
  - historical/literary/context expansions
- Personalize synthesis using cross-library context (current, read, want-to-read, and notes across books).

## Architecture Direction
- Session-first domain model (`recording -> transcribing -> synthesizing -> review -> complete|failed`).
- Provider-agnostic STT gateway with fallback policy.
- Multi-stage synthesis pipeline with schema validation and partial-failure tolerance.
- Review/publish step before synthesized artifacts become permanent notes.
- Strong observability for reliability + cost.

### Now (core foundation)
- [ ] #144 STT benchmark + provider decision matrix
- [ ] #145 Convex data model and session state machine
- [ ] #146 Recording UX on book detail page
- [ ] #147 Resilient audio capture and chunk upload
- [ ] #148 Session cap, alerts, and forced rollover

### Next (delivery slices)
- [ ] #149 STT gateway + provider adapters + fallback policy
- [ ] #150 Persist transcripts + auto-save raw transcript notes
- [ ] #151 Multi-stage synthesis pipeline + schema validation
- [ ] #152 Context packer for personalized synthesis
- [ ] #153 Review and publish workflow for synthesized artifacts
- [ ] #154 Observability + fallback telemetry + cost guardrails
- [ ] #155 Reliability test suite for long sessions + rollover

## Vox Patterns to Reuse
- Streaming session orchestration with bounded finalize timeout + fallback.
- Guardrail-first capture integrity checks.
- Provider fallback strategy with typed errors.

## Definition of Done
- Reader can complete: record -> transcribe -> synthesize -> review -> publish.
- Raw transcript is never lost.
- Cap transitions are explicit and safe.
- Synthesized outputs are contextually useful and traceable.

### LINKED ISSUE #148: [Listening Sessions] Enforce session cap with audible warning and forced rollover
## Parent
- Part of #105

## Problem
Processing caps are inevitable. Without explicit warning + rollover behavior, users lose trust.

## Scope
- Define and implement max session policy (configurable cap, e.g. 30/45/60 min).
- Add pre-cutoff alerts:
  - Audible alert (required).
  - Prominent visual warning.
- At cap: auto-stop current session, auto-trigger processing, and force explicit start of a new session.
- Handle background/locked-screen edge cases with best-effort user warning semantics.

## Acceptance Criteria
- [ ] User receives clear audible + visual warning before cutoff.
- [ ] Cap transition auto-processes current recording safely.
- [ ] New session cannot silently continue under old session ID.
- [ ] UX copy makes cap and rollover behavior obvious.
- [ ] Events are tracked (`cap_warning_shown`, `cap_reached`, `session_rollover_started`).

## Out of Scope
- Synthesis quality tuning.

### PR DESCRIPTION
**Pull request by phrazzld**
## What changed
- Added session cap telemetry events (`cap_warning_shown`, `cap_reached`, `session_rollover_started`) and explicit rollover UX cue (`Record next session`).
- Fixed rollover state safety so `capRolloverReady` is reset on processing failures.
- Added focused hook tests for rollover transitions and PostHog mocking in `components/notes/useListeningSessionRecorder.test.tsx`.
- Reset session-specific cap state (`capNotice`, `capRolloverReady`) when `bookId` changes to avoid cross-book carryover.

## Validation
- bun run lint
- bun run typecheck
- bun run test -- components/notes/useListeningSessionRecorder.test.tsx
- pre-push hooks: full `vitest run` + `next build`

## Before / After
**Before:** PR was blocked by unresolved review threads on rollover edge cases and missing coverage. Rollover state could persist after failed capped processing and could carry across `bookId` changes.

**After:** Rollover state is now reset on failure and on `bookId` changes, hook coverage was added for rollover and telemetry behavior, review threads were resolved, and all required CI checks are green.

Closes #105
Closes #148

## Summary by CodeRabbit

* **New Features**
  * Recording button label now updates to "Record next session" when a rollover is available after hitting a cap
  * Session rollover lets you resume recording in a fresh session once capacity is reached
  * Improved analytics capture for cap warnings, cap reached, and rollover starts
  * Rollover and cap notices are cleared when switching the tracked item

* **Tests**
  * Comprehensive test coverage added for listening session recording flows and edge cases

### CODE REVIEW

File: components/notes/useListeningSessionRecorder.ts [hunk 1]
```diff
@@ -128,6 +133,7 @@ export function useListeningSessionRecorder(bookId: Id<"books">) {
   const [lastProvider, setLastProvider] = useState<string>("");
   const [lastArtifacts, setLastArtifacts] = useState<SynthesisArtifacts | null>(null);
   const [capNotice, setCapNotice] = useState<string | null>(null);
+  const [capRolloverReady, setCapRolloverReady] = useState(false);
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply components/notes/useListeningSessionRecorder.ts]: Addressed in 5e55ef5. Added a `bookId` effect to reset `capNotice` and `capRolloverReady` in the hook, plus coverage in `components/notes/useListeningSessionRecorder.test.tsx` (`resets rollover and cap notice when book changes`).
